tests/MyGame/Example/Monster.java

```diff
@@ -47,6 +47,7 @@
   public Monster testarrayoftables(int j) { return testarrayoftables(new Monster(), j); }
   public Monster testarrayoftables(Monster obj, int j) { int o = __offset(26); return o != 0 ? obj.__assign(__indirect(__vector(o) + j * 4), bb) : null; }
   public int testarrayoftablesLength() { int o = __offset(26); return o != 0 ? __vector_len(o) : 0; }
+  public Monster testarrayoftablesByKey( String key ) { int vectorOffset = __vector(__offset(26)) - 4; return vectorOffset != 0 ? Monster.lookupByKey(bb.array().length - vectorOffset, key, bb) : null;  }
```

Comment:
Indeed, it seems the example I picked up for generation is wrong. 
I'll fix the check for missing field case, it is indeed buggy. 

I must say I don't understand lookupByKey() by key implementation:
```
    int vectorLocation = bb.array().length - vectorOffset; // why ??? 
    int span = bb.getInt(vectorLocation);
    int start = 0;
    vectorLocation += 4; // The reason for the -4, lookupByKey expect offset on data containing size of the array
```
AFAIK, __vector already provided a ready to use offset on the first item of the array. I don't understand why the byte buffer size is involved in computing vectorLocation...
```
  protected int __vector(int offset) {
    offset += bb_pos;
    return offset + bb.getInt(offset) + SIZEOF_INT;  // data starts after the length
  }
```
If I understood correctly, you want me to:
- remove the odd ` int vectorLocation = bb.array().length - vectorOffset;`
- change it so that it can directly work with the offset provided by __vector() (do the -4 inside to get the vector size)

This change is going to break all existing code on this public API. Is that ok?